nexus/tests/integration_tests/affinity.rs

```diff
@@ -755,6 +769,49 @@ async fn test_group_crud<T: AffinityGroupish>(client: &ClientTestContext) {
     assert!(groups.is_empty());
 }
 
+#[nexus_test]
+async fn test_affinity_instance_group_list(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let external_client = &cptestctx.external_client;
+    test_instance_group_list::<AffinityType>(external_client).await;
+}
+
+#[nexus_test]
+async fn test_anti_affinity_instance_group_list(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let external_client = &cptestctx.external_client;
+    test_instance_group_list::<AntiAffinityType>(external_client).await;
+}
+
+async fn test_instance_group_list<T: AffinityGroupish>(
+    client: &ClientTestContext,
+) {
+    const PROJECT_NAME: &'static str = "test-project";
+
+    let api = ApiHelper::new(client);
+
+    // Create an IP pool and project that we'll use for testing.
+    create_default_ip_pool(&client).await;
+    api.create_project(PROJECT_NAME).await;
+
+    let project_api = api.use_project::<T>(PROJECT_NAME);
```

Comment:
WHAT IS THIS BEAUTIFUL API

Reply:
a hackery of query parameter appending for the affinity tests... but feel free to re-use it, if you want!